src/engine/sys/sdl_glimp.cpp

```diff
@@ -101,6 +101,13 @@ static Cvar::Cvar<bool> r_ext_texture_integer( "r_ext_texture_integer",
 static Cvar::Cvar<bool> r_ext_texture_rg( "r_ext_texture_rg",
 	"Use GL_EXT_texture_rg if available", Cvar::NONE, true );
 
+static Cvar::Cvar<bool> workaround_noTextureGather_nvidia340( "workaround.noTextureGather.nvidia340", "Disable ARB_texture_gather on Nvidia 340 driver", Cvar::NONE, true );
```

Comment:
ISTM they should have `r_` prefix

Reply:
This is part of the discussion I want to have indeed.

I assume the hierarchical dot-spaced cvar names is “the new naming”, while `<prefix>_camelCase is “the legacy naming”. We still use the legacy naming a lot, mainly for consistency, like within the renderer (`r_*`), but it doesn't mean we cannot continue using new naming. I even myself suggested in the past we may implement cvar name aliases to be able to port all cvars to the dotted-hierarchical naming while retaining compatibility with old names in scripts (or in user's memory).

This workaround namespace is totally new, so there is no cargo cult attached with it, so it may sounds counter-productive to do it the legacy way if we consider the dotted-hierarchical way is the future. That `workaround.*` namespace is meant to live alongside existing `logs.*`, `common.*`, `vm.*`, `audio.*`, etc.

Now we can discuss the naming, is `workaround` the right top level name? Should it be named otherwise? Should it be child of another name?

Then comes the question about all those workarounds being about the renderer, so we may suggest `renderer.workaround.noTextureGather.nvidia340`.

But we may have other workarounds in the future for other things, like audio, input, network or filesystem and want all of them under the same tree for easy completion, so we may suggest `workaround.renderer.noTextureGather.nvidia340`.

In the end the extra parent (here `.renderer`) between the top level and the workaround name seems to be very noisy, I strongly doubt a possible audio, input, network or filesystem workaround name would collide with a renderer one, so I went with listing all workarounds under the `workaround` top level.

Something I want to remind is that we better not mix the naming schemes, likes using dots in a `r_` cvar, using underscores in a `common.` cvar, or using underscore as tree separator in place of the dot (i.e. no more than one underscore), I now some bot cvars did the latter but we better not want to do more and limit the amount of conventions to one legacy and one new and not mix them.

Reply:
I don't think it's a good idea to make workaround a top-level namespace. I think all renderer cvars is a more useful category than all workaround

Reply:
I think it's better to not pollute an existing namespace like `r_` and I actually like having a dedicated namespace for this.

One usage of those cvars will be to use them in bug reports, giving driver developers a way to reproduce a bug we have a workaround for. For example we would be able to tell a Mesa developer to do this to reproduce the R600 Hyper-Z bug:

```
./daemon -set workaround.noHyperZ.mesaRv600 off +devmap plat23
```

I like the way it looks, and it is easy to autocomplete as well.

Reply:
I'll reiterate the point I made on dev channel, here: workaround cvars would be out of place under `r_*`, because most of the time you won't use them. Either `workaround.renderer.*` or `renderer.workaround.*` would be better. The latter especially so, since it's very unlikely we'd have such workarounds for non-renderer things: any sort of OS bug or difference in handling something would most likely be handled at the compiler-level, and we wouldn't be able to anything about it at runtime, unlike avoiding graphics driver bugs.

Reply:
> I'll reiterate the point I made on dev channel, here: workaround cvars would be out of place under `r_*`, because most of the time you won't use them.

This point doesn't make sense to me as half of the cvars under `r_` are obscure debugging things already. The workaround stuff is at least something a player might want to use in principle (if the workaround misclassifies the hardware/driver as broken when it actually works fine).

Anyway, I agree that non-renderer workarounds will be extremely rare and so a renderer-based namespace is better. So let's go with `renderer.workaround`.

Reply:
So I renamed to `renderer.workaround`. 

I don't really see those cvars as ones a player may use, as the provided fallbacks are assumed to have more chance to work than what they are working around.

It's more thought as a developer things, both for engine developers and drivers developers: being able to test code path we intentionally don't provide to users by default on a given hardware or driver.